dotCMS/src/main/java/com/dotcms/util/DbExporterUtil.java

```diff
@@ -0,0 +1,237 @@
+import com.google.common.collect.ImmutableList;
+import com.zaxxer.hikari.HikariDataSource;
+import io.vavr.Lazy;
+import io.vavr.control.Try;
+import org.apache.commons.io.IOUtils;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import static java.io.File.separator;
+
+/**
+ * Utility class to export Postgres DB dump to a sql.gz file
+ *
+ * @author vic
+ */
+public class DbExporterUtil {
+    /**
+     * pg_dump binary distribution.
+     */
+    public enum PG_ARCH {
+        ARM("pg_dump_aarch64"), AMD64("pg_dump_amd64"), OSX_INTEL("pg_dump_x86_64");
+
+        final private String pgDump;
+
+        PG_ARCH(String pgDump) {
+            this.pgDump = pgDump;
+        }
+
+        public static String pgDump() {
+            if ("amd64".equalsIgnoreCase(System.getProperty("os.arch"))) {
+                return AMD64.pgDump;
+            }
+            if ("aarch64".equalsIgnoreCase(System.getProperty("os.arch"))) {
+                return ARM.pgDump;
+            }
+            if ("x86_64".equalsIgnoreCase(System.getProperty("os.arch"))) {
+                return OSX_INTEL.pgDump;
+            }
+            throw new DotRuntimeException("Unable to determine architecture for : " + System.getenv("os.arch"));
+        }
+    }
+    
+    /**
+     * Utility private and empty constructor.
+     */
+    private DbExporterUtil() {}
+
+    /**
+     * Executes pg_dump binary and stores its output in a generated SQL gzipped file.
+     *
+     * @return file with DB dump
+     */
+    public static File exportToFile() {
+        final String hostName = Try.of(() ->
+                        APILocator.getHostAPI()
+                                .findDefaultHost(
+                                        APILocator.systemUser(),
+                                        false)
+                                .getHostname())
+                .getOrElse("dotcms");
+        final String fileName = StringUtils.sanitizeFileName(hostName)
+                + "_db_"
+                + DateUtil.EXPORTING_DATE_FORMAT.format(new Date())
+                + ".sql.gz";
+        final File file = new File(ConfigUtils.getAbsoluteAssetsRootPath()
+                + separator
+                + "server"
+                + separator
+                + "sql_backups"
+                + separator
+                + fileName);
+        final long starter = System.currentTimeMillis();
+        Logger.info(DbExporterUtil.class, "DB Dumping database to  : " + file);
+        if (file.exists()) {
+            throw new DotRuntimeException("DB Dump already exists:" + file);
+        }
+        file.mkdirs();
+        file.delete();
```

Comment:
there are modern ways to do the make dir:
```
Files.createDirectories
```

Reply:
Switched.